libsql-server/src/namespace/mod.rs

```diff
@@ -1045,6 +1043,12 @@ impl Namespace<PrimaryDatabase> {
 
         let bottomless_replicator = if let Some(options) = &config.bottomless_replication {
             let options = make_bottomless_options(options, bottomless_db_id, name.clone());
+            if bottomless::replicator::Replicator::has_backup_of(&options).await? {
+                return Err(Error::Anyhow(anyhow!(
```

Comment:
does it make sense? Isn't make_new_primary called simply when we instantiate the namespace into memory? Then it would be just fine to also have an existing backup on s3, and it will usually be true. Or is make_new_primary only called for brand new namespaces?

Reply:
That was the question I asked @MarinPostma on Slack. I've wrote this piece while waiting for the response.

Reply:
It's called when it's loaded into memory, but before we'd also check if it was necessary to perform a restore. With the meta store that should not be necessary anymore, and we'd only check at namespace creation